### DISCUSSION TITLE
Saleor Rewind 2022 🍾

### DISCUSSION THREAD

**Discussion by peelar** [saleor/saleor#11608] [Announcements]
Hello everyone 👋 !

We wish you all the best in the upcoming year. We are ending this one with [Saleor Rewind](https://saleor.io/blog/saleor-rewind--154/) - an article summing up the most important features we have shipped in the last 12 months:

![image](https://user-images.githubusercontent.com/44495184/210097490-5fb20ba7-52e3-4c7b-b2e7-18def51af066.png)

https://saleor.io/blog/saleor-rewind--154/

What was your favorite feature released this year? What would you like to see added to Saleor in 2023?

Let us know in the comments! Happy New Year 🎇

**Comment by looped-monk** [comment]
Hey @peelar . Multi-channel is a fantastic feature.  What I would love to see next is a marketplace-multi-vendor feature which has features like amazon or maybe better which I believe suits a lot for startups worldwide. Would love to hear your views and plans regarding the same.

**Reply by peelar** [reply]
Thanks for your input @narayanan-ka 👋.

Multi-vendor is such a fundamental feature it is difficult to add it as an afterthought to a regular e-commerce API. I am afraid Saleor will not cut it here.

Can we add anything else to Saleor to make your life easier?

**Reply by looped-monk** [reply]
@peelar Thanks for the reply. Unfortunately in my use case, I'm  only looking for a marketplace setup built on the lines of Amazon because a pure inventory based model becomes difficult as a startup, from a investment perspective. I came across some repos like oscar, shuup  but I zeroed down to saleor only because it is having a apps based approach without modifying the core and uses graphql which are great. I just came across a repo which is building a multi vendor feature based on your repo : Saleor-multi-vendor(https://github.com/Saleor-Multi-Vendor/saleor-multi-vendor) which is a marketplace setup.Some key features are pending though. Thought I should just mention it here. I might have to rely on that and work my way. Thanks again.

**Reply by timuric** [reply]
@narayanan-ka multivendor is a very specific thing to each business, I am afraid there is no common denominator that would work for everyone. 

However, I think Saleor is a great fit for multivendor as it is, with some apps built on top of it. For example, we have users that simply build custom dashboards for their vendors, or those that create hundreds of independent websites using channels.

We would be happy to hear your ideas for generic features that would help model multivendor scenarios.

**Reply by looped-monk** [reply]
Hi@timuric .At the outset, I thank you for getting into this conversation. To emphasize furthur : My use case can be explained below. Kindly let me know if the existing setup of Saleor does this for me.

This is how a typical a marketplace should operate as a foundational structure. I would keep the fee structure charged by the marketplace for its services flexible which can be enabled/disabled based on each products requirement :

A marketplace is where Users can be Sellers(Vendors) themselves and have dedicated dashboard access to Saleor backend with permissions to only view, upload and sell their products.
The following needs to be carefully looked into : Order fulfilment will be in 3 options :
Complete Fulfilment by Us(My product firm) - Just like how Amazon does it (Fulfilled by Amazon)
The following will be the features:

Inventory management (Holding) in our Warehouse by us - Warehouse fee per sq. cubic feet.

Packaging done by us : Packaging fee per Box(Sizes:)

Delivery done by us - Delivery fee (0-250 gms, 251-500 gms)

Product listing on our platform : product listing fee per product.

Promotions management of Vendors product on our frontend app by us. Free.

Payment management - processing & refunds if applicable by us : Payment Processing fee.

Returns & Refunds management by us : Refund management fee.

Customer support will be handled by us : Support fee.
  

Partially fulfilled by Vendor :
The following will be the features:

Inventory management (Holding) by Vendor in his Warehouse. (Hence no fee by us charged)

Packaging done by us : Packaging fee per Box(Sizes:)

Delivery done by us - Delivery fee (0-250 gms, 251-500 gms etc).

Product listing on our platform : product listing fee per product.

Promotions management by Vendors themselves.

Payment management - processing & refunds if applicable by us : Payment Processing fee.

Returns & Refunds management by us : Refund management fee.

Customer support will be handled by us : Support fee.

Completely fulfilled by Vendor :
The following will be the features:

Inventory management (Holding) by Vendor in his Warehouse. (Hence no fee by us charged)

Packaging done by us : Packaging fee per Box(Sizes:)

Delivery done by us - Delivery fee (0-250 gms, 251-500 gms etc).

Product listing on our platform : product listing fee per product.

Promotions management by Vendors themselves.

Payment management - processing & refunds if applicable by vendor : Hence no Payment Processing fee.

Returns & Refunds management by Vendors : Hence No Refund management fee.

Customer support will be handled by us : Hence No Support fee.

The order is hence split into 3 categories above and the fee charged by marketplace platform. This is how typically a marketplace should work in a generic way. For a specific use case , there could be additions to the above but the core is the above. Howver, I just need to know with Saleor , can I achieve the above ?

**Reply by timuric** [reply]
It sounds like the only thing you need to provide a small subset of admin UI to vendors, for example applying promotions. You could use dashboard as reference when creating dashboard for vendors.

**Reply by looped-monk** [reply]
That is all since you mentioned the only thing ?

**Reply by leandrobortolotto** [reply]
Hello @timuric and @narayanan-ka. I am circling back here to see how the marketplace feature is going with Saleor. From what I understood from this discussion plus #10641 about marketplace, it looks like we can achieve it with the current Saleor features.

I would love to see an example of code (Frontend) that separates the products, settings, sales, refunds, payments, etc., based on the Vendor. In which a Vendor has access only to their own _products, settings, sales, refunds, payments, etc._ 

If you guys (@timuric @peelar and Saleor team) say it's possible, so probably it is.

On my side, we'll need to develop a marketplace that will sell products from other people, but they will send out the products. We'll only sell. The fulfillment will be done by them. We'll also sell digital products: an online course, a PDF, that doesn't require delivery. I think that should be fine, right?

I think Saleor is a great project, so I'll give it a try and see how we can improve it.

Please, enlighten me about anything, any company, any resource regarding a marketplace version for Saleor. It will be very helpful if we could have some examples to make the process easier.

Let me know if I can contribute, somehow.

**Reply by looped-monk** [reply]
@leandrobortolotto Hi. From what @timuric said, the only thing is required as per his recommendation is a minor change in a UI (per seller), which is a subset of the main dashboard ui. In the words of @timuric "It sounds like the only thing you need to provide a small subset of admin UI to vendors, for example applying promotions. You could use dashboard as reference when creating dashboard for vendors."

This is a reply that he sent me in response to my requirement of a marketplace (https://github.com/saleor/saleor/discussions/11608#discussioncomment-4593342)

**Comment by ghost** [comment]
In many open-source e-commerce projects, requesting support for multi-vendor functionality is common, and now that we are already in 2023, I hope the Saleor team can consider it. If Saleor can make it available out of the box, it will win more users.

**Reply by timuric** [reply]
@chrislais multi-vendor support is quite a broad topic, there are many stores using Saleor for marketplaces, please check this thread https://github.com/saleor/saleor/discussions/10641

In most cases by multi-vendor request people mean having a separate dashboard for vendors, which is in our view is the same as storefront functionality which needs to be built by the business. We provide an extensible ecommerce platform that makes no assumptions about how business interacts with vendors or customers via front-end layer. 

If you have in mind some specific marketplace features, please let us know.
